framework/caching/CRedisCache.php

```diff
@@ -85,13 +90,26 @@ class CRedisCache extends CCache
 	 */
 	protected function connect()
 	{
-		$this->_socket=@stream_socket_client(
-			$this->hostname.':'.$this->port,
-			$errorNumber,
-			$errorDescription,
-			$this->timeout ? $this->timeout : ini_get("default_socket_timeout"),
-			$this->options
+        if ($this->socket)
+        {
+            $this->_socket=@stream_socket_client(
+                "unix://$this->socket",
+                $errorNumber,
+                $errorDescription,
+                $this->timeout ? $this->timeout : ini_get("default_socket_timeout"),
+                $this->options
+            );
+        }
+        else
+        {
+            $this->_socket=@stream_socket_client(
+                $this->hostname.':'.$this->port,
+                $errorNumber,
+                $errorDescription,
+                $this->timeout ? $this->timeout : ini_get("default_socket_timeout"),
+                $this->options
 		);
+        }
```

Comment:
A lot of duplicate code is introduced here, the first parameter is the only one swapped.

Instead of introducing a new parameter we could also do:

```php
$address = $this->hostname;
if (isset($this->port)) {
    $address .= ':' . $this->port;
}

$this->_socket=@stream_socket_client(
    $address,
    $errorNumber,
    (...)
```

And then you specify it like:
```php
 'cache'=>array(
    'class'=>'CRedisCache',
    'hostname'=>'unix://run/redis/redis.sock',
    'port'=>null
),
```

Reply:
Passing socket path as `hostname` is extremely confusing. I suggest to solve this in the same way as in Yii2: 

https://github.com/yiisoft/yii2-redis/blob/f96b872fa646308ab2f11721a7d35c713a234621/src/Connection.php#L283